packages/pools/src/concentrated/fetch-pool-amount-provider.ts

```diff
@@ -0,0 +1,71 @@
+import { Int } from "@keplr-wallet/unit";
+
+import { AmountsDataProvider, ConcentratedLiquidityPool } from "./pool";
+
+type BankAmountResponse = {
+  balances: {
+    amount: string;
+    denom: string;
+  }[];
+};
+
+/** Provider of bank balances from bank module by bech32Address. Expected by concentrated liquidity pool. */
+export class FetchPoolAmountProvider implements AmountsDataProvider {
+  /** Address => Response */
+  protected amountsCache = new Map<
+    string,
+    Awaited<ReturnType<AmountsDataProvider["getPoolAmounts"]>>
+  >();
+
+  constructor(
+    protected readonly baseNodeUrl: string,
+    protected readonly poolId: string,
+    protected readonly cacheDurationMs = 1000 * 30, // 30 seconds
+    protected readonly amountFetcher: (
+      bech32Address: string
+    ) => Promise<BankAmountResponse> = async (bech32Address) => {
+      if (fetch === undefined)
```

Comment:
I built this with the intention of other teams using it. I think fetch API is ubiquitous at this point, but there could be some server side JS engine (perhaps an older version) that doesn't support it.

Reply:
Gotcha. In that case, despite adding to the bundle size, it might be better to use `axios` since axios will work across browser and nodejs environments, making the package more consistent/reliable

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#browser_compatibility

It's been supported in all browsers since 2015, almost 10 years.

Reply:
Your previous comment indicated that you were concerned about `fetch` not existing in some NodeJS/server side environments hence the `if (fetch === undefined)` check. It's difficult to know what percentage of devs are using Node 18+ (which includes `fetch`) but I would imagine a substantial portion are still using a version older than Node 18 (which would not have a `fetch` implementation). `axios` will ensure universal compatibility. 

If it only runs in browsers, I would not be concerned. The likelihood of someone running a browser that doesn't support `fetch` is very low.

Reply:
That's the reason the fetch function is dependency injected, to handle for that case where it's not supported in the server side VM.

For now, I think it's good to assume the modern option now, as it will only become stronger going forward.